player/player.go

```diff
@@ -0,0 +1,227 @@
+	if err != nil {
+		return err
+	}
+	s.DecodedKey = decodedKey
+	var blobsSizes string
+	for n, blob := range s.Blobs {
+		if blob.Length == stream.MaxBlobSize {
+			s.Size += blob.Length - 1
+		} else {
+			s.Size += blob.Length
+		}
+		decodedIV, err := hex.DecodeString(blob.IV)
+		if err != nil {
+			return err
+		}
+		s.Blobs[n].DecodedIV = decodedIV
+		if blobsSizes == "" {
+			blobsSizes = fmt.Sprintf("%v", blob.Length)
+		} else {
+			blobsSizes = fmt.Sprintf("%v+%v", blobsSizes, blob.Length)
+		}
+	}
+	// last padding is unguessable
+	s.Size -= 15
+	sort.Slice(s.Blobs, func(i, j int) bool { return s.Blobs[i].Number < s.Blobs[j].Number })
+	monitor.Logger.WithFields(log.Fields{
+		"blobs_number": len(s.Blobs),
+		"blobs_sizes":  blobsSizes,
+		"stream_size":  s.Size,
+		"uri":          s.URI,
+	}).Info("got stream data")
+	return nil
+}
+
+// prepareWriter writes necessary range headers and sets the status, preparing HTTP for partial blob streaming.
+// This should be called before streamBlobs
+func (s *reflectedStream) prepareWriter(startByte, endByte int64, writer http.ResponseWriter) (startBlob, endBlob int) {
+	if endByte == 0 {
+		endByte = int64(s.Size - 1)
+	}
+	startBlob = int(startByte / (stream.MaxBlobSize - 2))
+	endBlob = int(endByte / (stream.MaxBlobSize - 2))
+	rangeStart := startBlob * (stream.MaxBlobSize - 1)
+	rangeEnd := (endBlob + 1) * (stream.MaxBlobSize - 1)
+	if rangeEnd > s.Size {
+		rangeEnd = s.Size
+	}
+	resultingSize := rangeEnd + 1 - rangeStart
+	writer.Header().Set("Content-Type", s.ContentType)
+	writer.Header().Set("Accept-Ranges", "bytes")
+	writer.Header().Set("Content-Length", fmt.Sprintf("%v", resultingSize))
+	writer.Header().Set("Content-Range", fmt.Sprintf("bytes %v-%v/%v", rangeStart, rangeEnd, s.Size))
+	writer.WriteHeader(http.StatusPartialContent)
+	return startBlob, endBlob
+}
+
+// streamBlobs downloads, decodes and streams blobs `blobStart-blobEnd` into a given HTTP response
+func (s *reflectedStream) streamBlobs(blobStart, blobEnd int, writer http.ResponseWriter) error {
+	for _, blob := range s.Blobs[blobStart : blobEnd+1] {
+		monitor.Logger.WithFields(log.Fields{
+			"url": blob.URL(),
+		}).Info("requesting a blob")
+		httpRequest, err := http.NewRequest("GET", blob.URL(), new(bytes.Buffer))
+		httpClient := &http.Client{}
+		httpResponse, err := httpClient.Do(httpRequest)
+		if err != nil {
+			return err
+		}
+		defer httpResponse.Body.Close()
+		if httpResponse.StatusCode == http.StatusOK {
+			encryptedBody, err := ioutil.ReadAll(httpResponse.Body)
+			if err != nil {
+				return err
+			}
+			blob.encryptedBody = encryptedBody
+			decryptedBody, err := blob.decrypt(s)
+			if err != nil {
+				return err
+			}
+			writer.Write(decryptedBody)
+		} else {
+			return errors.Err("server responded with an unexpected status (%v)", httpResponse.Status)
+		}
+	}
+	return nil
+}
+
+func (b *reflectedBlob) decrypt(s *reflectedStream) (decryptedBody []byte, err error) {
```

Comment:
i added https://github.com/lbryio/lbry.go/commit/969a1423827178deed34ee77acfeb1e781f8f18d for you